### PR TITLE
implement 3 classes Carnivore/Herbivore/Animal

### PR DESCRIPTION
**Pull request by MaxSEngineer**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1 +1,43 @@
...
+    def bite(cls, obj: Herbivore) -> None:
+        for animal in super().alive:
+            if (
+                animal.name == obj.name
+                and isinstance(animal, Herbivore)
+                and not animal.hidden
+            ):
+                animal.health -= 50
+            if animal.health <= 0:
+                super().alive.remove(animal)
```
Comment (vsmutok) [inline app/main.py:43]:
```suggestion
    @staticmethod
    def bite(other_animal: Herbivore) -> None:
        if isinstance(other_animal, Herbivore) and not other_animal.hidden:
            other_animal.health -= 50
            other_animal.check_if_alive()
```

File: app/main.py [hunk 2]
```diff
@@ -1 +1,43 @@
...
+        )
+
+    def __str__(self) -> str:
+        return Animal.alive
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = True if not self.hidden else False
```
Comment (vsmutok) [inline app/main.py:28]:
```suggestion
        self.hidden = not self.hidden
```

**Review by vsmutok — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 1 above)
Comment (MaxSEngineer) [reply app/main.py]: Thanks, its really more clear code, i don't know why i used loop)

File: app/main.py (hunk 2 above)
Comment (MaxSEngineer) [reply app/main.py]: Makes sense so in short thanks

**Pull request closed without merging**